scripts/test

```diff
@@ -76,19 +76,28 @@ sleep 5
 while ! curl -sf http://localhost:8080/ping; do
     sleep 2
 done
+
 while ! kubectl --kubeconfig /etc/rancher/k3s/k3s.yaml rollout status -w -n cattle-system deploy/rancher-webhook; do
     sleep 2
 done
 #kill $TPID
 
+# get correct agent tag
```

Comment:
Can we do this in `scripts/version`?

Reply:
This needs to be fixed regardless as this wouldn't work when releases are tagged.

Reply:
So I decided to update `scripts/version` with this https://github.com/rancher/rancher/blob/9fcc2f6751e777531b5b5d33726e2f7577596eac/scripts/version#L4-L7

Reply:
Hmm, the implementation for `ETCD_UNSUPPORTED_ARCH` basically breaks part of the `scripts/version` script, which is the annoying part. Having this exception works but is not ideal as the whole check for `DIRTY` is that the repository has not been modified. If something modifies `package/Dockerfile` somewhere down the road, we won't be notified which is bad.

Reply:
So I updated the if statement to this instead https://github.com/rancher/rancher/blob/eb9d4f3c310a326763a78a1b5110e7ed0e7fb778/scripts/version#L4-L8
Now I check to see the `ETCD_UNSUPPORTED_ARCH` change instead.

Reply:
I appreciate the "cleaner" approach but I'd rather fix this properly and just set the environment variable if it is needed and avoid messing with local files (I know this is not your implementation). But I don't now with who this was discussed or what the deadline is for this PR

Reply:
Would this be a better solution? How about removing this conditional from rancher/scripts/package
https://github.com/rancher/rancher/blob/cc77991e41c3c4062ab115f1b57d817fe1c22fd5/scripts/package#L17-L22

and then putting this conditional in rancher/package/entrypoint.sh
```
if [ ${ETCD_ARCH} != amd64 ]; then
    export ETCD_UNSUPPORTED_ARCH=${ETCD_ARCH}
fi

```

where we set ETCD_ARCH in rancher/package/Dockerfile
`ENV ETCD_ARCH ${ARCH} `

Reply:
I'd prefer a combination of ARG and ENV like we partially have to avoid breaking changes in startup scripts. I believe having it empty by default and overriding it should cover both cases but obviously should be tested first.

Reply:
Ok so I updated the if statement 

https://github.com/igomez06/rancher/blob/336245d93df97fda5cfb8755a40d2f8f989c5d80/scripts/package#L17-L22

Plus now the docker build takes an `ETCD_UNSUPPORTED_ARCH` arg

https://github.com/igomez06/rancher/blob/336245d93df97fda5cfb8755a40d2f8f989c5d80/scripts/package#L24

And updated the docker file that will set `ETCD_UNSUPPORTED_ARCH` as an env var, it's an empty value by default

https://github.com/igomez06/rancher/blob/336245d93df97fda5cfb8755a40d2f8f989c5d80/package/Dockerfile#L17

https://github.com/igomez06/rancher/blob/336245d93df97fda5cfb8755a40d2f8f989c5d80/package/Dockerfile#L231